### PR TITLE
Add Buf Linter to Go Modules

### PR DESCRIPTION
**Pull request by danielsdeleo**
### :nut_and_bolt: Description: What code changed, and why?

Adds the buf linter to our go modules and lets you install the binary into the studio. This is the next step to running the linter on our protos in Ci.

### :chains: Related Resources

#4203 

### :athletic_shoe: How to Build and Test the Change

* Enter the studio
* `install_protoc_toolchain`
* `buf check lint`

There are a lot of errors/violations/whatevers when you run the linter, I will be getting to those in a future patch.

### CODE REVIEW

File: .studio/common [hunk 1]
```diff
@@ -31,8 +31,23 @@ function bootstrap_proto_compilers() {
...
+document "install_protoc_toolchain" <<DOC
+  Install all the protobuf plugins, linters, etc. we use to compile and vet our protobufs.
+DOC
+function install_protoc_toolchain() {
+  local proto_go_tools=(
+    github.com/chef/automate/components/automate-grpc/protoc-gen-policy
+    github.com/chef/automate/components/automate-grpc/protoc-gen-grpc-mock # generate mock grpc server implementations
+    github.com/chef/automate/components/automate-grpc/protoc-gen-a2-config
+    github.com/envoyproxy/protoc-gen-validate
+    github.com/ckaznocha/protoc-gen-lint
```
Comment (srenatus) [inline .studio/common:43]: I believe we can ditch this one if we're hooking up buf.

It's called here: https://github.com/chef/automate/blob/c9a5ab05e76c4fc4e91f0342caa5a16c1a431fb1/components/automate-gateway/scripts/grpc.sh#L86-L99

and we can remove the `nolint` comments from here, I suppose:
```console
$ git grep lint | grep proto:
api/config/gateway/config_request.proto:3:// nolint: 'V1' and 'S3' doesn't look like CamelCase to the linter. Curious.
components/automate-gateway/api/notifications/notifications.proto:3:// nolint: keeping this as-is for backwards-compat reasons (for now)
```

File: api/interservice/authn/authenticate.pb.validate.go [hunk 2]
```diff
@@ -33,6 +33,9 @@ var (
 	_ = ptypes.DynamicAny{}
 )
 
+// define the regex for a UUID once up-front
+var _authenticate_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")
+
```
Comment (srenatus) [inline api/interservice/authn/authenticate.pb.validate.go:38]: Huh this means we must have also upgraded our deps...? Not a bad thing. It means we could use the `uuid` well-known string type in a few places, maybe.

Comment (danielsdeleo) [reply api/interservice/authn/authenticate.pb.validate.go:38]: Yeah, installing `buf` via go modules requires pretty new versions of all the grpc/proto things, we should be completely up to date now 🎉

File: .studio/common (hunk 1 above)
Comment (danielsdeleo) [reply .studio/common:43]: 👍 thanks for digging in, I'll put this on my list for when I start actually running buf against the codebase

**Pull request merged**